packages/modules/devices/idm/idm/counter.py

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python3
+from typing import TypedDict, Any
+from pymodbus.constants import Endian
+
+from modules.common.abstract_device import AbstractCounter
+from modules.common.component_state import CounterState
+from modules.common.component_type import ComponentDescriptor
+from modules.common.fault_state import ComponentInfo, FaultState
+from modules.common.modbus import ModbusDataType, ModbusTcpClient_
+from modules.common.simcount import SimCounter
+from modules.common.store import get_counter_value_store
+from modules.devices.idm.idm.config import IDMCounterSetup
+
+
+class KwargsDict(TypedDict):
+    device_id: int
+    client: ModbusTcpClient_
+
+
+class IDMCounter(AbstractCounter):
+    def __init__(self, component_config: IDMCounterSetup, **kwargs: Any) -> None:
+        self.component_config = component_config
+        self.kwargs: KwargsDict = kwargs
+
+    def initialize(self) -> None:
+        self.__device_id: int = self.kwargs['device_id']
+        self.client: ModbusTcpClient_ = self.kwargs['client']
+        self.sim_counter = SimCounter(self.__device_id, self.component_config.id, prefix="bezug")
+        self.store = get_counter_value_store(self.component_config.id)
+        self.fault_state = FaultState(ComponentInfo.from_component_config(self.component_config))
+
+    def update(self):
+        unit = self.component_config.configuration.modbus_id
```

Comment:
Die Modbus-ID liegt in der Device config. Da über die Wärmepumpe wahrscheinlich keine weiteren Komponenten abgefragt werden können, ist es egal, ob die ModbusID in den Gerät oder Komponenten Einstellungen liegt.
Du kannst im kwargs dict noch die ModbusID ergänzen und dann mit an die Komponente übergeben.
Kannst Du das bitte noch ändern?